### ISSUE TITLE
CAMLEncodeLayerTreeToPathWithInfo produce unstable XML

### ISSUE THREAD

**Issue by paulz** [paulz/ViewSnapshots#12]
While all tests pass on M1 on https://github.com/paulz/ViewSnapshots/commit/ad1c88bb66b419460523b251c48ae1ef22850dbc

`matchesCABundle(ComplexView_Previews.self)` fails on Intel:

M1:
```xml
<contents tint="0 0 0" type="CATintedImage">
	<image A8asL8="1" src="assets/image4.png" />
```

Intel:
```xml
<contents type="CATintedImage">
	<image A8asL8="1" src="assets/image4.png" type="CGImage" />
	<tint type="CGColor" value="0 0 0" />
```

https://github.com/paulz/ViewSnapshots/commit/ad1c88bb66b419460523b251c48ae1ef22850dbc#diff-ae46e2d8ddae3dd55c523050a6d1977374c100a9d7b1ab5168a9846fb4eef633L62-R63

**Comment by paulz** [comment]
## Intel
### VisualTestKit version.plist
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>BuildAliasOf</key>
	<string>VisualTestKit</string>
	<key>BuildVersion</key>
	<string>6082</string>
	<key>CFBundleShortVersionString</key>
	<string>1.0</string>
	<key>CFBundleVersion</key>
	<string>1</string>
	<key>ProjectName</key>
	<string>VisualTestKit_iosmac</string>
	<key>SourceVersion</key>
	<string>5000000000000</string>
</dict>
</plist>
```
### VisualTestKit Info.plist
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>BuildMachineOSBuild</key>
	<string>20A241133</string>
	<key>CFBundleDevelopmentRegion</key>
	<string>en</string>
	<key>CFBundleExecutable</key>
	<string>VisualTestKit</string>
	<key>CFBundleIdentifier</key>
	<string>com.apple.VisualTestKit</string>
	<key>CFBundleInfoDictionaryVersion</key>
	<string>6.0</string>
	<key>CFBundleName</key>
	<string>VisualTestKit</string>
	<key>CFBundlePackageType</key>
	<string>FMWK</string>
	<key>CFBundleShortVersionString</key>
	<string>1.0</string>
	<key>CFBundleSupportedPlatforms</key>
	<array>
		<string>MacOSX</string>
	</array>
	<key>CFBundleVersion</key>
	<string>1</string>
	<key>DTCompiler</key>
	<string>com.apple.compilers.llvm.clang.1_0</string>
	<key>DTPlatformBuild</key>
	<string>13E6049a</string>
	<key>DTPlatformName</key>
	<string>macosx</string>
	<key>DTPlatformVersion</key>
	<string>12.4</string>
	<key>DTSDKBuild</key>
	<string>21F64</string>
	<key>DTSDKName</key>
	<string>macosx12.4.internal</string>
	<key>DTXcode</key>
	<string>1330</string>
	<key>DTXcodeBuild</key>
	<string>13E6049a</string>
	<key>LSMinimumSystemVersion</key>
	<string>10.15</string>
	<key>UIDeviceFamily</key>
	<array>
		<integer>2</integer>
		<integer>6</integer>
	</array>
</dict>
</plist>
```

**Comment by paulz** [comment]
## Github runner

### VisualTestKit version.plist
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>BuildAliasOf</key>
	<string>VisualTestKit</string>
	<key>BuildVersion</key>
	<string>6082</string>
	<key>CFBundleShortVersionString</key>
	<string>1.0</string>
	<key>CFBundleVersion</key>
	<string>1</string>
	<key>ProjectName</key>
	<string>VisualTestKit_iosmac</string>
	<key>SourceVersion</key>
	<string>5000000000000</string>
</dict>
</plist>
```

### VisualTestKit Info.plist
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>BuildMachineOSBuild</key>
	<string>20A241133</string>
	<key>CFBundleDevelopmentRegion</key>
	<string>en</string>
	<key>CFBundleExecutable</key>
	<string>VisualTestKit</string>
	<key>CFBundleIdentifier</key>
	<string>com.apple.VisualTestKit</string>
	<key>CFBundleInfoDictionaryVersion</key>
	<string>6.0</string>
	<key>CFBundleName</key>
	<string>VisualTestKit</string>
	<key>CFBundlePackageType</key>
	<string>FMWK</string>
	<key>CFBundleShortVersionString</key>
	<string>1.0</string>
	<key>CFBundleSupportedPlatforms</key>
	<array>
		<string>MacOSX</string>
	</array>
	<key>CFBundleVersion</key>
	<string>1</string>
	<key>DTCompiler</key>
	<string>com.apple.compilers.llvm.clang.1_0</string>
	<key>DTPlatformBuild</key>
	<string>13E6049a</string>
	<key>DTPlatformName</key>
	<string>macosx</string>
	<key>DTPlatformVersion</key>
	<string>12.4</string>
	<key>DTSDKBuild</key>
	<string>21F64</string>
	<key>DTSDKName</key>
	<string>macosx12.4.internal</string>
	<key>DTXcode</key>
	<string>1330</string>
	<key>DTXcodeBuild</key>
	<string>13E6049a</string>
	<key>LSMinimumSystemVersion</key>
	<string>10.15</string>
	<key>UIDeviceFamily</key>
	<array>
		<integer>2</integer>
		<integer>6</integer>
	</array>
</dict>
</plist>
```

**Comment by paulz** [comment]
fixed in cfd7159d95b1065ff231038cacde07132b5ab44c

**Comment by paulz** [comment]
Failed on M1:

[Test-ViewSnapshots-2022.07.03_00-28-42--0700.xcresult.zip](https://github.com/paulz/ViewSnapshots/files/9034259/Test-ViewSnapshots-2022.07.03_00-28-42--0700.xcresult.zip)

<img width="1079" alt="Screen Shot 2022-07-03 at 12 30 29 AM" src="https://user-images.githubusercontent.com/59230/177029863-1f28bb8b-768b-466a-99e5-61df01d0e4d3.png">

### actual
```xml
					<image A8asL8="1" src="assets/image4.png" />
					<tint tint="0 0 0" type="CGColor" />
```

### expected
```xml
					<image A8asL8="1" src="assets/image4.png" type="CGImage" />
					<tint type="CGColor" value="0 0 0" />
```
